pygeos/set_operations.py

```diff
@@ -16,45 +16,87 @@
 ]
 
 @multithreading_enabled
-def difference(a, b, **kwargs):
+def difference(a, b, grid_size=None, **kwargs):
     """Returns the part of geometry A that does not intersect with geometry B.
 
+    If grid_size is nonzero, input coordinates will be snapped to a precision grid of that
+    size and resulting coordinates will be snapped to that same grid.  If 0, this
+    operation will use double precision coordinates.
+
     Parameters
     ----------
     a : Geometry or array_like
     b : Geometry or array_like
+    grid_size : float, optional (default: None).
+        precision grid size; requires GEOS >= 3.9.0.
 
     Examples
     --------
+    >>> from pygeos.constructive import normalize
     >>> line = Geometry("LINESTRING (0 0, 2 2)")
     >>> difference(line, Geometry("LINESTRING (1 1, 3 3)"))
     <pygeos.Geometry LINESTRING (0 0, 1 1)>
     >>> difference(line, Geometry("LINESTRING EMPTY"))
     <pygeos.Geometry LINESTRING (0 0, 2 2)>
     >>> difference(line, None) is None
     True
+    >>> normalize(difference(box(0,0,2,2),box(1,1,3,3)))
+    <pygeos.Geometry POLYGON ((0 0, 0 2, 1 2, 1 1, 2 1, 2 0, 0 0))>
+    >>> normalize(difference(box(0.1,0.2,2.1,2.1),box(1,1,3,3), grid_size=1)) # doctest: +SKIP
```

Comment:
Is the doctest skip still needed with adding the `normalize`? (or if we keep skipping it, the normalize can be removed)

Reply:
We need the skip for GEOS < 3.9 (otherwise using `grid_size` raises exceptions); I did not find a way to conditionally enable some doctests based on GEOS version.

Reply:
> We need the skip for GEOS < 3.9 (otherwise using `grid_size` raises exceptions); I did not find a way to conditionally enable some doctests based on GEOS version.

Ah, yes of course. Skipping is fine then (we should maybe at some point consider to run the doctests only on a limited GEOS version range, eg only latest release). 

If we skip it, though, I would maybe still leave out the `normalize` to make the example easier to read.